### ISSUE TITLE
post_233604

### ISSUE THREAD

**Issue by cincuranet** [cincuranet/blog-comments#400]
imported 

https://www.tabsoverspaces.com/id/233604

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-03-30T19:45:42_:

<p>Probably some help with testing would be appreciated. Best to subscribe to the mailing list where I'll post some preview versions.</p>

**Comment by cincuranet** [comment]
Imported comment written by _wpostma_ on _2017-03-30T22:56:55_:

<p>Done.  firebird-net-provider subscribed.</p><p><a href="https://lists.sourceforge.net/lists/listinfo/firebird-net-provider" rel="nofollow noopener" title="https://lists.sourceforge.net/lists/listinfo/firebird-net-provider">https://lists.sourceforge.n...</a></p>

**Comment by cincuranet** [comment]
Imported comment written by _Kevin Mitchell_ on _2017-05-31T01:14:41_:

<p>I followed your instructions and it looks like Firebird is working in Visual Studio 2017 on Windows 10 Pro.</p><p>1. I installed DDEXProvider-3.0.2.0.msi<br>2. I installed the Firebird .NET provider, FirebirdSql.Data.FirebirdClient-5.9.1.0.msi<br>3. I edited the machine.config file for 64-bit. Although it was probably unnecessary, I also edited the 32-bit version.<br>4. I edited the Registry per your instructions.</p>

**Comment by cincuranet** [comment]
Imported comment written by _quest_ on _2017-07-11T03:54:17_:

<p>i was doing this before vs2017 , <br>but on vs 2017 stucked :(<br>is there a way every vs2017 open auto import this keys to privateregistry.bin</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-07-11T07:29:38_:

<p>Maybe scripting it, that should be doable in reasonable amount of time.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Hosathota Vishwanatha_ on _2017-07-25T16:50:52_:

<p>It looked straight forward when I read the instructions, but when I tried this, VS2017 is not listing the provider I have included. Is there anything I need to check?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Walter Antolini_ on _2017-08-07T20:00:46_:

<p>Hi, the guide working, but never having changed the private registry of VS, it would have been more convenient to have a guide. The docs at the link: <a href="https://visualstudioextensions.vlasovstudio.com/2017/06/29/changing-visual-studio" rel="nofollow noopener" title="https://visualstudioextensions.vlasovstudio.com/2017/06/29/changing-visual-studio">https://visualstudioextensi...</a> -2017-private-registry-settings /. Thanks.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Miro_ on _2017-08-09T08:46:29_:

<p>Hi. Just did all the things in the article. I can see Firebird  Data Source now and choose it. After entering the connection data "test Connection" is successful, but when I hit The Ok button I get this message<br>"Unable to find the requested .Net Framework Data Provider.  It may not be installed."<br>What can be the reason for that?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-08-09T09:34:55_:

<p>FirebirdClient is not installed or can't be properly loaded.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Miro_ on _2017-08-10T13:22:31_:

<p>It's installed with PMC. I'm even testing the new changes in FbRemoteEvent right now.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-08-10T14:14:09_:

<p>Study: <a href="https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/DDEX/readme.txt" rel="nofollow noopener" title="https://github.com/cincuranet/FirebirdSql.Data.FirebirdClient/blob/master/DDEX/readme.txt">https://github.com/cincuran...</a> .</p>

**Comment by cincuranet** [comment]
Imported comment written by _henoc salinas_ on _2017-08-29T15:50:50_:

<p>It works perfect, do not forget to release the tree before running Visual Studio again.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2017-11-22T14:32:38_:

<p>You should do `gacutil /l FirebirdSql.Data.FirebirdClient`, you're working with assembly names not files at that point.</p>

**Comment by cincuranet** [comment]
Imported comment written by _José Carlos De Oliveira Junior_ on _2017-11-22T14:40:41_:

<p>**********************************************************************<br>** Visual Studio 2017 Developer Command Prompt v15.4.4<br>** Copyright (c) 2017 Microsoft Corporation<br>**********************************************************************</p><p>C:\Users\junio\source>gacutil /i "C:\Program Files (x86)\FirebirdClient\FirebirdSql.Data.FirebirdClient.dll"<br>Microsoft (R) .NET Global Assembly Cache Utility.  Version 4.0.30319.0<br>Copyright (c) Microsoft Corporation.  All rights reserved.</p><p>Assembly successfully added to the cache</p><p>C:\Users\junio\source>gacutil /l "C:\Program Files (x86)\FirebirdClient\FirebirdSql.Data.FirebirdClient.dll"<br>Microsoft (R) .NET Global Assembly Cache Utility.  Version 4.0.30319.0<br>Copyright (c) Microsoft Corporation.  All rights reserved.</p><p>The Global Assembly Cache contains the following assemblies:<br>Failure enumerating assemblies: Invalid file or assembly name.  The name of the file must be the name of the assembly plus .dll or .exe .</p><p>Number of items = 0</p>

**Comment by cincuranet** [comment]
Imported comment written by _Mario Sifuentes_ on _2018-03-13T22:37:15_:

<p>I have tryied to work with EF and Firebird, but I get an error each time when I was trying to setup a FB connection. I have follow all the instructions to work with Visual Studio 2017 Community and Visual Studio 2015 Community, but in both versions I got the same error. Basically, when I try to configure the screen of "Firebird Data source", the application crashes when I write in one of the fields or when I select an item of the combo boxes that are in that screen.<br>Can someone help me to solve this error?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2018-03-14T08:39:08_:

<p>That usually means your GAC and machine.config do not match.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Alex_ on _2018-03-26T17:33:19_:

<p>I have the same problem. Did you fix it?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Marcos Perez_ on _2018-04-06T20:33:30_:

<p>hi, i get "Failure has occurred while loading a type." in connection properties when trying to add a connection.<br>What should i do?</p>

**Comment by cincuranet** [comment]
Imported comment written by _Mario Sifuentes_ on _2018-04-11T16:32:13_:

<p>Thank you. I still wothout a solution, but your answer is appreciated.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Rafał Staniewski (Rafal612)_ on _2018-04-20T15:25:23_:

<p>Works! But when I trying add new connection VS2017 crasching and restarting. Smoeone had this problem ?<br>Edit: Don't have any erros just crasching and restarting.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Dimi_ on _2018-04-30T20:05:39_:

<p>I'm stuck in the same condition. It works fine, but when I try to select database or do anything on de configuration screen, VS2017 crashes.</p>

**Comment by cincuranet** [comment]
Imported comment written by _SJFriedl_ on _2018-06-10T06:10:31_:

<p>Thank you for providing all this; it's obviously a lot of work.</p><p>A few notes on the blog post itself. The two files mentioned in:</p><p>> Take a copy of FirebirdDDEXProvider64.reg (or FirebirdDDEXProvider32.reg)</p><p>are identical - no 32-vs-64 bit differences that I could see - and they both refer to version 14, not 15: they refer to HKEY_CURRENT_USER rather than the private registry in VS2017.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2018-06-10T07:15:55_:

<p>Not sure what you are talking about. The change is mentioned following the "...and change the registry paths...".</p>

**Comment by cincuranet** [comment]
Imported comment written by _Xavier_ on _2018-08-24T16:47:09_:

<p>I've applied all the settings described here on your website, however when I try to add the connection setting the window closes unexpectedly. What can it be? <a href="https://uploads.disquscdn.com/images/e71b3bf5465221ba963464f6e7c625f1d32d3fa495ba2bd51436f50d62269301.jpg" rel="nofollow noopener" title="https://uploads.disquscdn.com/images/e71b3bf5465221ba963464f6e7c625f1d32d3fa495ba2bd51436f50d62269301.jpg">https://uploads.disquscdn.c...</a></p>

**Comment by cincuranet** [comment]
Imported comment written by _Krzysztof Jozwiak_ on _2018-09-10T22:10:35_:

<p>I was able to solve this by installing FirebirdSql.Data.FirebirdClient-6.3.0.msi</p>

**Comment by cincuranet** [comment]
Imported comment written by _Olivier Dufour_ on _2018-09-11T12:12:57_:

<p>Hi Xavier!</p><p>I think you're not far to solve your problem :)</p><p>Check your "machine.config" files (If you have any doubt, check 32 AND 64 bits versions).<br>For example, on my machine, they are located :<br>- Here : C:\Windows\Microsoft.NET\Framework\v4.0.30319\Config<br>- And here : C:\Windows\Microsoft.NET\Framework64\v4.0.30319\Config</p><p>Be sure that there's only one firebird entry in section <a href="http://system.data" rel="nofollow noopener" title="system.data">system.data</a> --> DbProviderFactories</p><p>Please keep the entry whose version match your referenced assembly (For example 5.1.0.0, 6.1.0.0, 6.3.0.0, ...)</p><p>Hope this helps!</p>

**Comment by cincuranet** [comment]
Imported comment written by _Choudhry_ on _2018-11-10T03:30:50_:

<p>Hi there,</p><p>Need some help. It works through Server Explorer <br> <a href="https://uploads.disquscdn.com/images/6ab35bfecebe421bbed9f8fe0bc2dc0cfd59c8a47ca74a95b509fe8d09ddf0c3.png" rel="nofollow noopener" title="https://uploads.disquscdn.com/images/6ab35bfecebe421bbed9f8fe0bc2dc0cfd59c8a47ca74a95b509fe8d09ddf0c3.png">https://uploads.disquscdn.c...</a></p><p>But Not through when creating Entity Data Model Wizard > New Connection <a href="https://uploads.disquscdn.com/images/66e2b1a4859895d6598a01ca4afa58b6f62a33f96d85f4602dc8b95140470956.png" rel="nofollow noopener" title="https://uploads.disquscdn.com/images/66e2b1a4859895d6598a01ca4afa58b6f62a33f96d85f4602dc8b95140470956.png">https://uploads.disquscdn.c...</a></p><p>Any help would be appreciated.</p><p>Thanks.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2018-11-14T08:35:22_:

<p>It's still planned (see <a href="http://tracker.firebirdsql.org/browse/DNET-659)" rel="nofollow noopener" title="http://tracker.firebirdsql.org/browse/DNET-659)">http://tracker.firebirdsql....</a>. But it's all about priorities and my available time.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Jiří Činčura ↹_ on _2019-06-26T08:53:09_:

<p>Yes. It does.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Lehko Peter_ on _2019-07-20T17:50:04_:

<p>Hi Xavier<br>It is hapenning to me as well. As soon as I start editing the connection, it closes.</p>

**Comment by cincuranet** [comment]
Imported comment written by _leeand_ on _2019-10-14T05:12:12_:

<p>It became clear that VS2017 does overwrite the registry every now and then. Then all the successful changes are once again lost and need to be redone ( merged into the loaded HIVE...). From the Microsoft forum the only definite solution is to create an VS installer which seams doable as they give a detailed example (Oracle). Still too steep of a learning curve to climb for me. A script to merge the custom reg file into binary registry file would be a good alternative. Still waiting on this.</p>

**Comment by cincuranet** [comment]
Imported comment written by _Choudhry_ on _2019-12-08T06:50:18_:

<p>Hi, Trying in Visual Studio 2019 Enterprise. I get the first dialog bu the second dialog does not show the connection builder instead it is showing some connection string builder dialog <a href="https://uploads.disquscdn.com/images/43d6f3e1ddaf45d6f15cea597a6726c99afc9be9f879489c2a62bc7fca160a77.png" rel="nofollow noopener" title="https://uploads.disquscdn.com/images/43d6f3e1ddaf45d6f15cea597a6726c99afc9be9f879489c2a62bc7fca160a77.png">https://uploads.disquscdn.c...</a>  <a href="https://uploads.disquscdn.com/images/20fee27f1bd9f58e2b60372b1b364e6013c69f17d52db458c507907ccde1ada6.png" rel="nofollow noopener" title="https://uploads.disquscdn.com/images/20fee27f1bd9f58e2b60372b1b364e6013c69f17d52db458c507907ccde1ada6.png">https://uploads.disquscdn.c...</a></p>

**Comment by cincuranet** [comment]
Imported comment written by _Hzuniga_ on _2020-04-22T01:24:32_:

<p>Buenas solucionaste yo tambien llegue hasta alli</p>

**Comment by cincuranet** [comment]
Imported comment written by _Yomar Guerra_ on _2020-09-25T07:31:33_:

<p>Me puedes ayudar, necesito instalarlo en Visual studio 2019?</p>

**Comment by InOcobo** [comment]
How to install it in Visual Studio 2022?

**Comment by cincuranet** [comment]
Same steps, basically. But DDEX is no longer maintained, so I wouldn't use it.

**Comment by InOcobo** [comment]
> Same steps, basically. But DDEX is no longer maintained, so I wouldn't use it.

So, if the recommendation is not to use it...

What tool do I install to be able to add a data source in Visual Studio that connects to a FireBird database?

**Comment by cincuranet** [comment]
I would either use pure ADO.NET or abstraction like EF Core.

**Comment by InOcobo** [comment]
> I would either use pure ADO.NET or abstraction like EF Core.

But then, I wouldn't be able to use the wizards in VS 2022 that automatically generate the datasets and tableadapter for me with their queries.
